Uploader/upload/TeamscaleUpload.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+using NLog;
+
+/// <summary>
+/// Uploads trace files to Teamscale.
+/// </summary>
+internal class TeamscaleUpload : IUpload
+{
+    private static readonly Logger logger = LogManager.GetCurrentClassLogger();
+
+    private readonly HttpClient client;
+
+    private readonly TeamscaleServer server;
+    private readonly MessageFormatter messageFormatter;
+
+    public TeamscaleUpload(TeamscaleServer server)
+    {
+        this.server = server;
+        this.messageFormatter = new MessageFormatter(server);
+
+        this.client = new HttpClient();
+        SetUpBasicAuthentication(server);
+    }
+
+    private void SetUpBasicAuthentication(TeamscaleServer server)
+    {
+        byte[] byteArray = Encoding.ASCII.GetBytes($"{server.Username}:{server.AccessToken}");
+        client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
+    }
+
+    /// <summary>
+    /// Performs the upload asynchronously.
+    /// </summary>
+    /// <param name="filePath">Path to the file to upload.</param>
+    /// <param name="version">The application version (read from a version assembly).</param>
+    /// <returns>Whether the upload was successful.</returns>
+    public async Task<bool> UploadAsync(string filePath, string version)
+    {
+        logger.Debug("Uploading {tracePath} with version {version} to {teamscale}", filePath, server.ToString(), version);
+        using (MultipartFormDataContent content = new MultipartFormDataContent("Upload----" + DateTime.Now.Ticks.ToString("x")))
```

Comment:
i think the c# WebClient class provide a UploadFileAsync method

Reply:
I looked at that. TS unfortunately requires that the POST parameter be named "report" that contains the file. The WebClient doesn't let me control this, so I think that won't work :-(